syntax = "proto3";

package syft.grid.messages;

import "proto/core/common/common_object.proto";
import "proto/core/io/address.proto";

// CREATE Dataset
message CreateDatasetMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  bytes dataset = 3;
  map<string, string> metadata = 4;
  syft.core.io.Address reply_to = 5;
  string platform = 6;
}

// GET Dataset
message GetDatasetMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  int64 dataset_id = 3;
  syft.core.io.Address reply_to = 4;
}

message GetDatasetResponse {
  syft.core.common.UID msg_id = 1;
  map<string, string> metadata = 2;
  syft.core.io.Address address = 3;
}

// GET ALL Datasets
message GetDatasetsMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 3;
}

message GetDatasetsResponse {
  message metadata_container {
    map<string, bytes> metadata = 1;
  }
  syft.core.common.UID msg_id = 1;
  repeated metadata_container metadatas = 2;
  syft.core.io.Address address = 3;
}

// UPDATE Dataset
message UpdateDatasetMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  int64 dataset_id = 3;
  map<string, string> metadata = 4;
  syft.core.io.Address reply_to = 5;
}

// UPDATE Dataset
message DeleteDatasetMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  int64 dataset_id = 3;
  syft.core.io.Address reply_to = 4;
}
